starter-code/pizza.js

```diff
@@ -1 +1,67 @@
-// Write your Pizza Builder JavaScript in this file.
+$(document).ready(function(){
+    
+    //sauce and crust removal by default
+    $('.crust').removeClass('crust-gluten-free');
+    $('.sauce').removeClass('sauce-white');
+    
+    //sauce and crust button off by default
+    $('.btn-sauce').removeClass('active');
+    $('.btn-crust').removeClass('active');
+    
+    //hide crust and sauce items on the list and update the free one
+    $('li:contains("gluten-free crust")').text("0$ gluten-free crust");
+    $('li:contains("white"), li:contains("gluten-free")').hide();
+    
+    //events
+    clickEvent('.btn-pepperonni', '.pep', null, 'pepperonni');
+    clickEvent('.btn-mushrooms', '.mushroom', null, 'mushrooms');
+    clickEvent('.btn-green-peppers', '.green-pepper', null, 'green peppers');
+    clickEvent('.btn-sauce', '.sauce', 'sauce-white', 'white sauce');
+    clickEvent('.btn-crust', '.crust', 'crust-gluten-free', 'gluten-free crust');
+    
+    updatePrice();
+    
+});
+
+function clickEvent(selector, toggle, toggleClass, keyword){
+    
+    $(selector).click(function(){
+        $(this).toggleClass('active');
```

Comment:
Usas `this` aquí y un poco más abajo, no está mal, pero el callback recibe como parámetros diferentes elementos. En el caso del `click`, la función recibe el evento, y a partir del evento puedes obtener lo mismo que con el `this`. En la siguiente función también utilizas `this`. El callback de `each` en jQuery recibe como parámetros `index` y `element`; y con `element` tienes lo mismo que con `this`. Intenta utilizar lo que te proporciona el entorno en el que estás. Te lo digo porque puedes coger vicio con `this` sin conocer lo que ya te proporcionan las funciones